### PR TITLE
Move config and backend variables

### PR DESCRIPTION
**Pull request by a-sor**
The variables `config` and `backend` have been moved to `serles/__init__.py`, so that configuration can be accessed globally using `serles.config`, and also to avoid parsing configuration twice.

### CODE REVIEW

**Comment by uedvt359** [comment]
this looks fairly sensible at first glance. i'm going to need to test this a bit before merging it though.

**Comment by uedvt359** [comment]
looking at this again, i think there's some stuff missing. you didn't touch `init_config()` (here and in other files) at all. this should go away now, right?

**Comment by a-sor** [comment]
Yes, my bad, sorry :) Please take a look at the updated version. The tests also seem to work:
```
$ make -f tests/Makefile
PYTHONPATH=.:./lib coverage3 run -m pytest tests
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.10.12, pytest-8.1.1, pluggy-1.5.0
rootdir: /tmp/serles-acme-patched
collected 77 items

tests/test_appfactory.py .                                                                                                                                        [  1%]
tests/test_certbotbackend.py .......                                                                                                                              [ 10%]
tests/test_challengefuns.py .......................                                                                                                               [ 40%]
tests/test_configparser.py ....                                                                                                                                   [ 45%]
tests/test_ejbcabackend.py .....                                                                                                                                  [ 51%]
tests/test_flaskfuns.py ................                                                                                                                          [ 72%]
tests/test_helpers.py ........                                                                                                                                    [ 83%]
tests/test_opensslbackend.py ..                                                                                                                                   [ 85%]
tests/test_views.py ...........                                                                                                                                   [100%]

=========================================================================== warnings summary ============================================================================
tests/test_certbotbackend.py::CertbotBackendTester::test_backend_okconfig
  /tmp/serles-acme-patched/tests/test_certbotbackend.py:45: DeprecationWarning: Please use assertNotEqual instead.
    self.assertNotEquals(backend, None)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
===================================================================== 77 passed, 1 warning in 1.97s =====================================================================
coverage3 html --omit=serles/__main__.py,serles/backends/base.py serles/*.py serles/backends/*.py
Wrote HTML report to htmlcov/index.html
```

File: serles/challenge.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import requests
 import jwcrypto.jwk  # fedora package: python3-jwcrypto.noarch
 import jwcrypto.jws
+import serles
```
Comment (uedvt359) [inline serles/challenge.py:8]: can't we do `from . import config, backend`?

File: tests/test_challengefuns.py [hunk 2]
```diff
@@ -147,8 +148,8 @@ def mockedDNSResolve(qname, rdtype, search=False):
 
 class ChallengeFunctionTester(unittest.TestCase):
     def setUp(self):
-        main.backend = MockBackend.Backend([])
-        main.config = {
+        serles.backend = MockBackend.Backend([])
+        serles.config = {
```
Comment (uedvt359) [inline tests/test_challengefuns.py:152]: why the rename? i'd rather not have unnecessary changes

**Comment by a-sor** [comment]
Let me offer you a different approach. We could just cache the return values of `get_config()`, so that it doesn't parse the configuration repeatedly if called multiple times. This way, the differences from the original code are [minimal](https://github.com/dvtirol/serles-acme/pull/5/files). Please take look at the patch.

**Comment by uedvt359** [comment]
hi, thanks! sorry for taking so long, but i finally merged your contribution.

i locally ran the test suite and (after an unrelated fix) it still passes.
